bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -612,6 +629,51 @@ public void asyncReadUnconfirmedEntries(long firstEntry, long lastEntry, ReadCal
         asyncReadEntriesInternal(firstEntry, lastEntry, cb, ctx);
     }
 
+    /**
+     * Read a sequence of entries asynchronously.
+     *
+     * @param firstEntry
+     *          id of first entry of sequence
+     * @param lastEntry
+     *          id of last entry of sequence
+     */
+    @Override
+    public CompletableFuture<Iterable<org.apache.bookkeeper.client.api.LedgerEntry>> read(long firstEntry, long lastEntry) {
```

Comment:
this is public API, it is confusing to have so many read calls

readEntries
asyncReadEntries
read (and now)

1) naming of this method doesnt sound appropriate. It is not saying exactly how is it different from other read calls.
2) why are we making these changes in this pull request / commit, probably it would be better to limit this commit to builder interface of create/open/delete operations and create separate PR/commit for these ledgerhandle changes

Reply:
I know but LedgerHandle is a total mess. We are going to have a new clean API.
As we are introducing a new API it is the case to create an API which covers the full client side API. During the discussion on this proposal we decided to drop the callback based API and the byte[] APIs for instance. 
We do not want to create a new LedgerHandle implementation which will be a risk.
We could create a subclass of LedgerHandle which adds the read(), append(), write() methods but it will be a mess anyway and I do not see a real value

Reply:
@reddycharan with this new api, we are encouranging people using the new api. the old api will be deprecated in 4.7.